File: order-service/src/main/java/com/intexsoft/webshop/orderservice/dto/status/StatusDto.java
package com.intexsoft.webshop.orderservice.dto.status;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.intexsoft.webshop.orderservice.model.enums.StatusName;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class StatusDto {
    Long id;
    @JsonFormat(pattern = "dd-MM-yyyy HH:mm:ss")
    LocalDateTime createdAt;
    @Enumerated(EnumType.STRING)
    StatusName statusName;
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/dto/status/StatusDto.kt
package com.intexsoft.webshop.orderservicekt.dto.status

import com.fasterxml.jackson.annotation.JsonFormat
import com.intexsoft.webshop.orderservicekt.model.enums.StatusName
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import java.time.LocalDateTime

data class StatusDto(
    var id: Long,
    @JsonFormat(pattern = "dd-MM-yyyy HH:mm:ss")
    val createdAt: LocalDateTime,
    @Enumerated(EnumType.STRING)
    val statusName: StatusName,
)
